packages/grafana-ui/src/components/Combobox/Combobox.test.tsx

```diff
@@ -223,6 +229,28 @@ describe('Combobox', () => {
       jest.clearAllTimers();
     });
 
+    it('should debounce requests', async () => {
+      const asyncSpy = jest.fn();
+      const asyncOptions = jest.fn(async () => {
+        return new Promise<ComboboxOption[]>(asyncSpy);
+      });
+
+      render(<Combobox options={asyncOptions} value={null} onChange={onChangeHandler} />);
+
+      const input = screen.getByRole('combobox');
+      await user.click(input);
+
+      expect(asyncSpy).toHaveBeenCalledTimes(1); // Called on open
+      asyncSpy.mockClear();
+
+      await user.keyboard('abc');
+      await act(async () => {
+        jest.advanceTimersByTime(300); // Debounce value
+      });
```

Comment:
maybe this could demonstrate the functionality a bit more if it was like

```
await user.keyboard('a');
await act(async () => jest.advanceTimersByTime(10));

await user.keyboard('b');
await act(async () => jest.advanceTimersByTime(10));

await user.keyboard('c');
await act(async () => jest.advanceTimersByTime(300));

expect(asyncSpy).toHaveBeenCalledTimes(1);
```

thoughts?